### PR TITLE
Conform to spec when generating PromiseRejectionEvent

### LINKED ISSUE #1447: PromiseRejectionEvent differs from Living Standard spec
1) What version of bluebird is the issue happening on?
3.5.0

2) What platform and version? (For example Node.js 0.12 or Google Chrome 32)
Google Chrome 56.0

3) Did this issue happen with earlier version of bluebird?
Yes.

Bluebird seems to be creating a PromiseRejectionEvent with a 'detail' property, which contains 'promise' and 'reason' properties. TypeScript complains that PromiseRejectionEvent has no 'detail' property, which prompted me to look at the specification for this event. It appears that 'promise' and 'reason' are intended to be properties of the event object itself, e.g. event.promise and event.reason.

Perhaps I'm missing something?

Spec info found here -
https://developer.mozilla.org/en-US/docs/Web/API/PromiseRejectionEvent/PromiseRejectionEvent
https://html.spec.whatwg.org/multipage/webappapis.html#promiserejectionevent

### PR DESCRIPTION
**Pull request by nornagon**
Fixes #1447

### CODE REVIEW

File: src/debuggability.js [hunk 1]
```diff
@@ -156,7 +156,9 @@ var fireDomEvent = (function() {
             return function(name, event) {
                 var domEvent = new CustomEvent(name.toLowerCase(), {
                     detail: event,
-                    cancelable: true
+                    cancelable: true,
+                    reason: event.reason,
+                    promise: event.promise
```
Comment (nornagon) [inline src/debuggability.js]: Would it make sense to write this as `get promise() { return event.promise; }`? I'm unfamiliar with the compilation pipeline for bluebird, and whether that syntax is supported on all the browsers Bluebird targets.

If we wanted to be even more protective, we could do something like:
```js
var reason = event.reason;
var promise = event.promise;
var domEvent = new CustomEvent(name.toLowerCase(), {
  detail: event,
  cancelable: true,
  get reason() { return reason; },
  get promise() { return promise; }
})
```

or the equivalent with `Object.defineProperty`

Let me know if that's a desirable change & I can update the PR

**Comment by nornagon** [comment]
I just realised this doesn't handle the case where CustomEvent isn't defined, will update.

File: src/debuggability.js (hunk 1 above)
Comment (benjamingr) [reply src/debuggability.js]: Bluebird actually targets ES3 browsers too (supports IE7 for example) - on the other hand this branch of code isn't reached on these browsers anyway.

**Comment by nornagon** [comment]
Updated to use getters. Let me know if I should squash these commits.

File: src/debuggability.js [hunk 2]
```diff
@@ -154,12 +154,17 @@ var fireDomEvent = (function() {
...
+                var promise = event.promise;
+                var reason = event.reason;
+                var eventData = {
                     detail: event,
-                    cancelable: true,
-                    reason: event.reason,
-                    promise: event.promise
+                    cancelable: true
+                };
+                Object.defineProperties(eventData, {
```
Comment (benjamingr) [inline src/debuggability.js]: If you're using this syntax anyway - can you please use the helpers from util that work in ES3 environments for consistency?

Comment (nornagon) [reply src/debuggability.js]: Done (I think!)

**Comment by petkaantonov** [comment]
so is it supposed to be data property or getters? it's currently data property

**Comment by petkaantonov** [comment]
checked the spec and its readonly data.. merging

**Pull request merged**